Add option_detailed_parse config value.

marxin opened the pull request:
Support parsing of "variable part" for option directives similarly
to samp roles.

Fixes: #9965.

tk0miya committed:
extlinks: Emit a warning if extlnks config does not contain '%s'

To let users know the deprecation surely, use logger.warning instead
of RemovedInSphinxXXWarning since 5.0. It will cause a crash if users
use `-W` option on their CI.

tk0miya committed:
Bump version

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge pull request #9085 from tk0miya/8898_emit_warning_for_extlinks

extlinks: Emit a warning if extlnks config does not contain '%s'

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge pull request #9329 from tk0miya/master

Merge 4.x to master

shimizukawa committed:
using new issue template feature in yaml configuration

shimizukawa committed:
Update and rename bug_report.yml to bug-report.yml

tk0miya committed:
Merge branch '4.x'

astrojuanlu committed:
Document Python version policy

See https://groups.google.com/g/sphinx-dev/c/oDJ0KOLAmGk/m/X6N_UxpABAAJ

tk0miya committed:
Merge pull request #9396 from astrojuanlu/document-python-support

Document Python version support

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge pull request #9406 from shimizukawa/master

Migrate GitHub issue form with using new bug-report.yml

astrojuanlu committed:
Fix link to NumPy Docstring Standard

tk0miya committed:
Merge pull request #9425 from astrojuanlu/fix-numpy-docstring-link

Fix link to NumPy Docstring Standard

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

Commit:
:manpage: is a role, not a directive. [ci skip]

tk0miya committed:
Merge pull request #9503 from Un1Gfn/patch-1

doc: :manpage: is a role, not a directive

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

vedranmiletic committed:
Updated EXAMPLES

- removed or replaced missing docs
- removed docs that switched to using other tools
- added some new docs

tk0miya committed:
Merge pull request #9541 from vedranmiletic/documentation-fixes

Updated EXAMPLES

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

ndelon committed:
Fix typo in autodoc documentation

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge pull request #9581 from ndelon/fix-autodoc-typo

Fix typo in autodoc documentation

tk0miya committed:
refactor: linkcheck: Remove objects marked as RemovedInSphinx50Warning

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge pull request #9613 from tk0miya/RemovedInSphinx50Warning_linkcheck

refactor: linkcheck: Remove objects marked as RemovedInSphinx50Warning

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

lamby committed:
Make util.typing.restify sanitise unreproducible output (eg. memory addresses)

Whilst working on the Reproducible Builds effort [0] I noticed that sphinx
generates output that is not reproducible, causing a number of packages in
Debian to unreproducible.

Specifically, when Sphinx locates an alias of an instance when generating
'autodoc' documentation, it uses the raw Python repr(...) of the object and
does not sanitise it for memory addresses (etc.) like elsewhere in Sphinx.

This can result in documentation like this:

-<dd><p>alias of &lt;webob.client.SendRequest object at 0x7fd769189df0&gt;</p>
+<dd><p>alias of &lt;webob.client.SendRequest object at 0x7f0f02233df0&gt;</p>

Patch attached that uses the object_description method, which was added to fix
precisely this kind of issue.

I originally filed this in Debian as bug #996948 [1].

 [0] https://reproducible-builds.org/
 [1] https://bugs.debian.org/996948

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge pull request #9755 from lamby/996948-sphinx-please-make-the-output-of-instance-aliases-reproducible

Make util.typing.restify sanitise unreproducible output

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

jakobandersen committed:
Merge branch '4.x'

jakobandersen committed:
Change default value of intersphinx_disabled_reftypes

See discussion in sphinx-doc/sphinx#9459

jenshaase committed:
Fixed docker links in the installation.rst

The old links, linked to a login page of docker hub. The new links link to the public pages.

jakobandersen committed:
Merge pull request #9804 from jakobandersen/intersphinx_disabled_reftypes_default

Change default value of intersphinx_disabled_reftypes

tk0miya committed:
Merge pull request #9806 from jenshaase/patch-1

Fixed docker links in the installation.rst

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Fix merging error

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

gschwaer committed:
feat: add new option for autodoc_typehints_description_target to include undocumented return values

gschwaer committed:
feat: add tests and fix None rtype

gschwaer committed:
docs: add new option to documentation

spaceofmiah committed:
Fixes statement typo

tk0miya committed:
Merge pull request #9937 from spaceofmiah/patch-1

Fixes statement typo

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge pull request #9954 from tk0miya/master

Merge 4.x to master

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Deprecate sphinx.writers.latex.LaTeXWriter.docclasses

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
refactor: Remove RemovedInSphinx50Warning (partially)

tk0miya committed:
Postpone the removal of c_warn_on_allowed_pre_v3 warnings to v6.0

tk0miya committed:
Merge branch 'master' into RemovedInSphinx50Warning

tk0miya committed:
Merge pull request #9966 from tk0miya/RemovedInSphinx50Warning

refactor: Remove RemovedInSphinx50Warning (partially)

tk0miya committed:
Merge pull request #9978 from tk0miya/deprecate_LaTeXWriter.docclasses

Deprecate sphinx.writers.latex.LaTeXWriter.docclasses

tk0miya committed:
Fix #9962: texinfo: Do not use @definfoenclose to empasize string

Since texinfo-6.8, `@definfoenclose` command has been deprecated. This
stops to use it to reduce the deprecation warning.

This also disables the customization the styles of emphasized text via
the command.

jfbu committed:
LaTeX: separate terms from their definitions by a CR (fix #9985)

marxin committed:
Add option_detailed_parse config value.

Support parsing of "variable part" for option directives similarly
to samp roles.

Fixes: #9965.

jfbu committed:
Fix a spelling typo in tests (s/änhlich/ähnlich)

tk0miya committed:
Merge pull request #10006 from jfbu/German_spelling_typo_in_test

Fix a spelling typo in tests (s/änhlich/ähnlich)

tk0miya committed:
Merge pull request #9991 from tk0miya/9962_texinfo_definfoenclose

Fix #9962: texinfo: Do not use @definfoenclose to empasize string

jfbu committed:
Merge branch 'master' into 9985_latex_on_master

tk0miya committed:
Merge branch '4.x'

jfbu committed:
Merge branch 'master' into 9985_latex_on_master

jfbu committed:
Merge pull request #9999 from jfbu/9985_latex_on_master

LaTeX: separate terms from their definitions by a CR (close #9985)

jfbu committed:
Update CHANGES for PR #9999

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
autosummary: autosummary: import_by_name() now raises ImportExceptionGroup

It starts to `ImportExceptionGroup` exception instead of `ImportError` by default
when it failed to import target object.

tk0miya committed:
autodoc: The default of autodoc_typehints_format becomes to 'smart'

The default value of autodoc_typehints_format configuration is changed
to `'smart'`.  It will suppress the leading module names of typehints
(ex. `io.StringIO` -> `StringIO`).

refs: #9075

tk0miya committed:
CHANGES: Fix the location of the deprecation entry

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Close #9595: Deprecate setuptools integration

tk0miya committed:
Merge pull request #10038 from tk0miya/update_changes

CHANGES: Fix the location of the deprecation entry

tk0miya committed:
Merge pull request #10036 from tk0miya/9555_autosummary_raise_ImportExceptionGroup_by_default

autosummary: autosummary: import_by_name() now raises ImportExceptionGroup

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge pull request #10037 from tk0miya/9075_default_autodoc_typehints_format

autodoc: The default of autodoc_typehints_format becomes to 'smart'

tk0miya committed:
Merge branch 'master' into 9595_deprecate_setuptools_integration

pradyunsg committed:
Change "Permalink to this {headline -> heading}"

"heading" is a better word to use in this tooltip, since it better
matches the use case.

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge pull request #10048 from pradyunsg/change-heading-anchor-tooltip

Change "headline" to "heading"

tk0miya committed:
Merge pull request #10040 from tk0miya/9595_deprecate_setuptools_integration

Close #9595: Deprecate setuptools integration

not-my-profile committed:
doc: Linkify issue numbers in changelog to GitHub

tk0miya committed:
Merge pull request #10063 from not-my-profile/linkify-issues-in-changelog

doc: Linkify issue numbers in changelog to GitHub

tk0miya committed:
Close #10062: Change the default language to 'en'

Change the default language to `'en'` if any language is not set in
`conf.py`.

tk0miya committed:
Merge branch '4.x'

flying-sheep committed:
Fix typing in docfields.py

tk0miya committed:
Merge pull request #10084 from flying-sheep/patch-1

Fix typing in docfields.py

tk0miya committed:
Merge pull request #10067 from tk0miya/10062_change_default_language_to_en

Close #10062: Change the default language to 'en'

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

marxin commented:
I think the pull request is not correct. The difference I see based on the new config value:

```
Args:  directory
samp_role.parse: [<#text: ' directory'>]
desc_addname: <desc_addname classes="sig-prename descclassname" xml:space="preserve"> directory</desc_addname>
```

@tk0miya Can you please help me?

marxin commented:
> @tk0miya Can you please help me?

I think I understand it now and hope I have a proper solution that fixes cases where whitespaces were lost in HTML output.

AA-Turner committed:
Drop JavaScript Frameworks (#10028)

* Delete jQuery and underscore.js

* Move underscores.js setup to searchtools.js

* Update jQuery.url(en|de)code

* Update jQuery.getQueryParameters

* Firefox bug is no longer present

xref https://bugzilla.mozilla.org/show_bug.cgi?id=645075#c49

* Update jQuery.fn.highlightText

* Use enum instead of magic number

* Update test descriptions to remove obsolete jQuery reference

* Update Documentation.getCurrentURL

* Revert accidental fix of Documentation.getCurrentURL

* Update Documentation.initOnKeyListeners

* Update Documentation.hideSearchWords

* Update Documentation.highlightSearchWords

* Update Documentation.initDomainIndexTable

* Use arrow functions and const

* Replace $(document).ready

* Strict mode

* Move Documentation.hideSearchWords next to Documentation.highlightSearchWords

* Update translation functions in Documentation

* Replace $(document).ready in searchtools.js

* Update Scorer

* Update Search.hasIndex, Search.deferQuery, Search.stopPulse

* Prefer window.location

* Update Search.init

* Update Search.loadIndex

* Update Search.setIndex

* Update Search.startPulse

* Add _escapeRegExp

* Update Search.makeSearchSummary

* Update Search.htmlToText

* Update Search.performSearch

* Factor out _displayNextItem

* Update Search.query

* Update Search.performObjectSearch

* Update Search.performTermsSearch

* Remove underscores.js setup

* Use Sets

* Update test configuration

* Fix test failures

* Drop unused make/get URL functions

* Strict mode in searchtools.js

* Remove outmoded check for jQuery and underscore.js

* Ran prettier

prettier --print-width 120 --no-semi --quote-props as-needed --no-bracket-spacing --arrow-parens avoid --write sphinx/themes/basic/static

* Remove more references to jQuery and underscore.js

* Remove jQuery and underscore.js licences

* Update classic theme for no jQuery

* Update all other themes for no jQuery

* Restore jQuery & underscores.js to Sphinx themes

Enables a more gradual deprecation

* Added deprecation note to CHANGES

* Run prettier with defaults

* Update deprecation message to include extensions, note that sources must be copied

* oops

* Address Pradyun's feedback

* Forgot this one

* `let` doesn't work, as it is scoped to the block...

* Remove missed jQuery in sphinx13 theme

stsewd committed:
Searchtools: don't assume that all themes define some elements

When retrieving a non-existent element, jQuery would still return an
object (kind of empty one, so method calls won't raise a null
exception), but now `getElementById` will return null and raise an
exception when trying to call a method on that value.

This mainly affects the rtd theme,
which completely overrides the search page
https://github.com/readthedocs/sphinx_rtd_theme/blob/d64dadf1ceec4f9ff6c1ca2d3ea4c3d0fdb0e8d2/sphinx_rtd_theme/search.html

tk0miya committed:
CI: Test with Windows Server 2019

In nearly days, `windows-latest` image on GHA was changed to Windows
Server 2022.  That causes build errors on building a cython script.

To avoid the problem, this pins the image for testing on GHA to Windows
Server 2019.

stsewd committed:
Update sphinx/themes/basic/static/searchtools.js

Co-authored-by: Adam Turner <9087854+AA-Turner@users.noreply.github.com>

jfbu committed:
LaTeX: correct footnote marks, extended with page of link target

Fix #10188

Footnotes in some LaTeX environments (tables, fulllineitems for object
descriptions) are gathered and appear after the environment, causing the
footnote to possibly appear on a page later than some of the footnote
marks referring it.

With this commit, the footnote mark compares page numbers and
incorporates the destination page number if it turns out to be distinct
from the page where it stands.

tk0miya committed:
Merge commit '2be06309518d9401a42880bb5b4321dfdd1e5e90'

tk0miya committed:
Merge commit '07110b7557a552ecfe702bdd6a2d2f9685cb1af9'

jfbu committed:
LaTeX: comment out footnote mark enhancement for footnote environment

Page detection left to apply as was case before this PR, i.e. only for
the \sphinxfootnotemark mark-up, which means either in combination
with footnotetext environment, or for multiply referred-to footnotes
and the extra footnote marks additional to the initial one.

This is to avoid a problem with each LaTeX pass in certain circumstances
modifying the aux file, and latexmk consequently complaining.

refs:
https://github.com/sphinx-doc/sphinx/pull/10191#issuecomment-1038807448

jfbu committed:
LaTeX: avoid a footnote in a signature line executing \label twice

The signature may contain an auto-generated footnote, which now
always sets a label (formerly this happened only for multiply
referred-to footnotes so the problem did not arise).

jfbu committed:
LaTeX: workaround a memoir class compatibility issue

tk0miya committed:
Merge commit '799385f5558a888d1a143bf703d06b66d6717fe4'

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge pull request #10207 from tk0miya/master

Merge 4.x to master

carltongibson committed:
Fixed typo.

Should read `…an info field list…`, rather than ``…a info field list…`.

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge pull request #10217 from carltongibson/patch-1

Fixed typo.

stsewd committed:
Searchtools: fix null query on setIndex

Previous to https://github.com/sphinx-doc/sphinx/pull/10028
this code was https://github.com/sphinx-doc/sphinx/blob/b60caca9408bb778ec0b277eed105dab8cfae0bb/sphinx/themes/basic/static/searchtools.js#L93-L100

The current code is making _queued_query null,
and then doing a search with that instead of doing the search
with the previous value.

tk0miya committed:
Merge pull request #10233 from stsewd/fix-set-index

Searchtools: fix null query on setIndex

calebchiam committed:
fixing duplicate search results bug

calebchiam committed:
Merge branch 'sphinx-doc:master' into master

jakobandersen commented:
> > @tk0miya Can you please help me?
> 
> I think I understand it now and hope I have a proper solution that fixes cases where whitespaces were lost in HTML output.

In the ``allowable_client`` example, what is the HTML around the whitespace?
For future-proofing it might be good to render a space explicitly by a ``addnodes.desc_sig_space()``.

tk0miya committed:
Merge pull request #10236 from calebchiam/master

Fixing duplicate search results bug (5.0)

tk0miya committed:
Update CHANGES for PR #10236

marxin commented:
> In the `allowable_client` example, what is the HTML around the whitespace?

Looks like this:

```html
<span id="cmdoption-git-commit-allowable-client"></span><span class="sig-name descname"><span class="pre">-allowable_client</span></span><span class="sig-prename descclassname"> </span><em><span class="pre">client_name</span></em><a class="headerlink" href="#cmdoption-git-commit-allowable_client" title="Permalink to this definition">¶</a></dt>
```

> For future-proofing it might be good to render a space explicitly by a `addnodes.desc_sig_space()`.

Do you render this before and after a "variable" part?

jakobandersen commented on sphinx/domains/std.py:
```diff
@@ -202,7 +204,16 @@ def handle_signature(self, sig: str, signode: desc_signature) -> str:
+                    if isinstance(part, nodes.Text):
+                        # in order to preserve whitespaces, call desc_addname
+                        text = part.astext()
+                        signode += addnodes.desc_addname(text, text)
```
I guess here one could do something like
```py
if len(text) != 0 and len(text.strip()) == 0:
   signode += addnodes.desc_sig_space()
else:
   signode += addnodes.desc_addname(text, text)
```
but it is more a theoretical exercise at this point I guess.

jakobandersen commented:
> > In the `allowable_client` example, what is the HTML around the whitespace?
> 
> Looks like this:
> 
> ```
> <span id="cmdoption-git-commit-allowable-client"></span><span class="sig-name descname"><span class="pre">-allowable_client</span></span><span class="sig-prename descclassname"> </span><em><span class="pre">client_name</span></em><a class="headerlink" href="#cmdoption-git-commit-allowable_client" title="Permalink to this definition">¶</a></dt>
> ```
> 
> > For future-proofing it might be good to render a space explicitly by a `addnodes.desc_sig_space()`.
> 
> Do you render this before and after a "variable" part?

I'm not completely sure what you mean, but the HTML that in your quote is ``<span class="sig-prename descclassname"> </span>`` would instead be generated by the ``addnodes.desc_sig_space()``, see inline code comment.

marxin commented:
> I'm not completely sure what you mean, but the HTML that in your quote is `<span class="sig-prename descclassname"> </span>` would instead be generated by the `addnodes.desc_sig_space()`, see inline code comment.

All right, I've just done that.

stsewd committed:
doctools: fix hideSearchWords

Previous code was:

https://github.com/sphinx-doc/sphinx/blob/b3812f72a98b01bae4b1158761082edc46cfa87f/sphinx/themes/basic/static/doctools.js#L262-L268

Ref https://github.com/sphinx-doc/sphinx/pull/10028

stsewd committed:
doctools: fix shortcuts

Previous code was using JQuery to add the listeners
https://github.com/sphinx-doc/sphinx/blob/b3812f72a98b01bae4b1158761082edc46cfa87f/sphinx/themes/basic/static/doctools.js#L304

Looks like by convention returning false mean preventDefault,
but now with pure js this needs to be called explicitly.

And a couple of fixes with the selectors.

- querySelectorAll is an array-like object, it doesn't have a first()
  method.
- nextLink was being assigned the `href` attribute instead of the
  object itself.

AA-Turner committed:
Update splitQuery

AA-Turner committed:
Regenerate jssplitter.py

tk0miya committed:
Merge pull request #10244 from stsewd/fix-hide-search

doctools: fix hideSearchWords

tk0miya committed:
Merge pull request #10245 from stsewd/fix-shortcuts

doctools: fix shortcuts

jbms committed:
[C++] Support conditional operator "?"

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch 'master' into fix-searchtools

tk0miya committed:
Merge pull request #10247 from AA-Turner/fix-searchtools

Fix query splitting in `searchtools.js`; update `jssplitter`

AA-Turner committed:
Fix object search

AA-Turner committed:
Replace `jssplitter` with JavaScript

tk0miya committed:
Merge pull request #10273 from AA-Turner/fix-object-search

`searchtools.js`: Fix object search, simplify `splitQuery`

Commit:
Fixed singledispatch documentation

Commit:
Removed white space

jakobandersen commented on sphinx/domains/std.py:
```diff
@@ -202,7 +204,19 @@ def handle_signature(self, sig: str, signode: desc_signature) -> str:
+                            signode += addnodes.desc_sig_space()
+                        else:
+                            # in order to preserve whitespaces, call desc_addname
+                            signode += addnodes.desc_addname(text, text)
```
Hmm, is ``desc_addname`` appropriate? In the "normal" programming language domains it is used for additional name parts, i.e., qualification of the main declared entity. But here it is not really for typesetting any name part.
Now that we are anyway changing this: to be conservative I suggest using the original ``nodes.Text`` until something a more specific markup is deemed appropriate.

marxin replied:
Hm, then I would need to find a way how to preserve whitespaces for a `nodes.Text` node.
E.g. for:

```
.. option:: -o directory, --object-directory directory, --object-file file
```

parts are: `_directory`, `_directory`, `_file`, where `_` is a space. If I don't do it, I would end up with:
![Screenshot from 2022-03-23 10-36-23](https://user-images.githubusercontent.com/2658545/159668708-9623fb51-f573-488e-ad74-2520a1873034.png)

jakobandersen replied:
How about these changes: https://github.com/jakobandersen/sphinx/tree/marxin-option_parse_variable_part

jakobandersen committed:
C++, use Itanium mangling for conditional expr

jakobandersen committed:
C++, var naming style in cond expr

jakobandersen committed:
C++, simplify assginemnt-expression parsing

jakobandersen committed:
Update CHANGES

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Merge branch '4.x'

tk0miya committed:
Fix #10214: html: invalid language tag was generated for zh_CN

So far, HTML builder output the `language` configuration as a language
tag for HTML.  But it takes locale string in ANSI C, not IETF language
code.

This converts locale string to language tag to generate valid language
tag for HTML.

Commit:
Closes #10310: Sphinx automodule: class method not documented when decorated with mocked function.

Commit:
Use recursive call to `ismock()` on unbound method instead of using temporary object.

marxin commented:
@tk0miya May I please ping this pull request? Thanks.

Commit:
Add unwittingly removed empty line.

Commit:
Add types and docstring to function `_method_is_bound()`.

tk0miya committed:
Fix #10279: autodoc: Default values are rendered as a string literal

When processing overloaded functions, autodoc unexpectedly renders its
default values for kwonlyargs as a string literal unexpectedly

jakobandersen commented on doc/usage/configuration.rst:
```diff
@@ -700,6 +700,14 @@ General configuration
+
+   Default is ``False``.
+   Similarly to ``samp`` role, support parsing of "variable part" of a literal text.
+   For example, in ``..option: -foption={TYPE}``,
```
```suggestion
   For example, in ``.. option:: -foption={TYPE}``,
```

jakobandersen commented on doc/usage/configuration.rst:
```diff
@@ -700,6 +700,14 @@ General configuration
+   For example, in ``..option: -foption={TYPE}``,
+   the part ``TYPE`` would be emphasized.
+
+   .. versionadded:: 4.5
```
```suggestion
   .. versionadded:: 5.0
```

jakobandersen commented:
I suggest making the conf var name more generic, e.g., ``option_detailed_parse``, and then change the docs to describe that it in general makes Sphinx parse the options more in detail to provide better markup, which includes the parsing of ``{stuff}`` as if by the ``:samp:`` role to signify variables, and that the curly braces can be escaped, etc.

Commit:
Rename function `_method_is_bound()` -> `isboundmethod()`.

Commit:
Move method `isboundmethod()` into sphinx/util/inspect.py.

Commit:
Use `safe_getattr()` to access `__self__`.

Commit:
Extend `test_mock_decorator` for class method.

Commit:
Reduce complexity by removing recursion.

marxin commented:
Made the suggested change to the config value name. Feel free to suggest wording improvements for the config value.

tk0miya committed:
Merge pull request #10315 from tk0miya/10279_defvalue_for_kwonlyargs_in_overloads

Fix #10279: autodoc: Default values are rendered as a string literal

tk0miya committed:
Fix #10280: autodoc_docstring_signature generates needless return typehint

Basically, autodoc suppresses return value typehint for class
constructors.  But it was unexpectedly shown if
`autodoc_docstring_signature` is enabled and docstring has multiple
signatures.

tk0miya committed:
Fix #10266: autodoc_preserve_defaults crashes on kwonlyargs w/o defaults

tk0miya committed:
Fix #10318: literalinclude: :prepend: option working with :dedent:

It will always be warned because prepending is processed before
dedenting.  This changes the order of processing options.

tk0miya committed:
Merge branch '4.x' into 5.x

tk0miya committed:
i18n: Change translation target branch from 4.x to 5.x

tk0miya committed:
Merge pull request #10324 from tk0miya/gha_transifex

i18n: Change translation target branch from 4.x to 5.x

tk0miya committed:
Merge pull request #10311 from tk0miya/10124_invalid_language_tag

Fix #10214: html: invalid language tag was generated for zh_CN

tk0miya committed:
Close #8417: autodoc: :inherited-members: option now takes multiple classes

It allows to suppress inherited members of several classes on the module at
once by specifying the option to `automodule` directive

tk0miya committed:
Update message catalogs

tk0miya committed:
Merge pull request #10327 from sphinx-doc/bot/pull-translations

Update message catalogs

tk0miya committed:
Merge pull request #10320 from tk0miya/10280_return_typehints_in_docstrings

Fix #10280: autodoc_docstring_signature generates needless return typehint

tk0miya committed:
Merge branch '5.x' into 10266_autodoc_preserve_defaults_crashes_on_kwonlyargs_without_defaults

tk0miya committed:
Merge pull request #10313 from GutiCW/feature-support_mocked_decorator_for_class_methods

Support mocked decorator for class methods

tk0miya committed:
Merge pull request #10321 from tk0miya/10266_autodoc_preserve_defaults_crashes_on_kwonlyargs_without_defaults

Fix #10266: autodoc_preserve_defaults crashes on kwonlyargs w/o defaults

tk0miya committed:
Update CHANGES for PR #10313

tk0miya committed:
Merge pull request #10323 from tk0miya/10318_literalinclude_prepend_and_dedent

Fix #10318: literalinclude: :prepend: option working with :dedent:

tk0miya committed:
Merge pull request #10325 from tk0miya/8417_inherited_members_takes_list_of_classes

Close #8417: autodoc: :inherited-members: option now takes multiple classes

tk0miya committed:
Merge branch '5.x' into autodoc_force_undocumented_rtype

tk0miya committed:
autodoc_typehints_description_target: Rename returnvalue_and_documented_params to documented_params

tk0miya committed:
Fix a testcase

tk0miya committed:
Merge branch '5.x' into autodoc_force_undocumented_rtype

tk0miya committed:
Merge pull request #9792 from gschwaer/autodoc_force_undocumented_rtype

add option for autodoc to hide undocumented parameters but show undocumented return types

Commit:
Added test case

tk0miya committed:
Fix a testcase

tk0miya committed:
Fix a testcase for py36

tk0miya committed:
Merge pull request #10285 from jmp1985/master

Fixed singledispatch documentation

tk0miya committed:
Update CHANGES for PR #10285

tk0miya committed:
Merge pull request #10153 from stsewd/check-if-search-progress-exists

Searchtools: don't assume that all themes define some elements

AA-Turner committed:
Fix highlighting

AA-Turner committed:
Run prettier

AA-Turner committed:
Fix test_restify

tk0miya committed:
Update message catalogs

tk0miya committed:
Merge pull request #10338 from sphinx-doc/bot/pull-translations

Update message catalogs

tk0miya committed:
Merge pull request #10336 from AA-Turner/fix-test-restify

Fix `test_restify`

tk0miya committed:
Merge pull request #10335 from AA-Turner/fix-highlighting-div-body

Fix search highlighting

tk0miya committed:
Merge branch '4.x' into 5.x

jfbu committed:
Update CHANGES for PR #10191

AA-Turner committed:
Don't use deprecated `setup.py build_sphinx`

jfbu committed:
Merge pull request #10191 from jfbu/10188_latexfootnotes

LaTeX: fix footnote marks for multiple references

AA-Turner committed:
Enable the opt-in EncodingWarning

https://docs.python.org/3/library/io.html#io-encoding-warning

AA-Turner committed:
Move `compile_catalog_plusjs` to `utils/`

AA-Turner committed:
Add command runners

AA-Turner committed:
Update documentation

AA-Turner committed:
Update transifex.yml

AA-Turner committed:
Remove deprecated code for Sphinx 5.0

tk0miya committed:
Update message catalogs

tk0miya committed:
Merge pull request #10358 from sphinx-doc/bot/pull-translations

Update message catalogs

tk0miya committed:
Merge pull request #10355 from AA-Turner/compile_catalogues

Move `babel` support to a dedicated utility file

tk0miya committed:
Merge branch '5.x' into sphinx-docs-build

tk0miya committed:
Merge pull request #10352 from AA-Turner/sphinx-docs-build

Don't use deprecated `setup.py build_sphinx`

tk0miya committed:
Change RemovedInSphinx60Warning to a subclass of DeprecationWarning

AA-Turner committed:
Review comments

AA-Turner committed:
Merge remote-tracking branch 'origin/remove-deprecated-50' into remove-deprecated-50

tk0miya committed:
Merge branch '5.x' into encoding-warnings

tk0miya committed:
Merge pull request #10357 from AA-Turner/remove-deprecated-50

Remove deprecated code for Sphinx 5.0

tk0miya committed:
Merge pull request #10354 from AA-Turner/encoding-warnings

Enable the opt-in EncodingWarning

jakobandersen commented:
> Made the suggested change to the config value name. Feel free to suggest wording improvements for the config value.

With the name change I suggest pulling in some variation of the changes I made in https://github.com/jakobandersen/sphinx/tree/marxin-option_parse_variable_part.

jakobandersen committed:
C++, bump env version for cond expr

jakobandersen committed:
Merge pull request #10302 from jakobandersen/cpp_cond_expr

C++, conditional expression (update of #10270)

shimizukawa committed:
fix typo

from https://github.com/sphinx-doc/sphinx-doc-translations/pull/13
